www/apps/docs/content/deployments/server/deploying-on-aws-using-terraform.md

```diff
@@ -0,0 +1,241 @@
+| Notification Module Providers | Amazon SES | *Planned for future implementation* |
+
+---
+
+## Step-by-Step Guide to Using the Terraform Module
+
+### Step 1: Download or Clone the Module
+
+Clone the Terraform module repository to your local machine:
+
+```shell
+git clone https://github.com/u11d-com/terraform-aws-medusajs
+cd terraform-u11d-medusajs/examples/minimal
+```
+
+### Step 2: Provide Variables for the Module
+
+See module inputs documentation for available configuration options.
+[Example configurations](https://registry.terraform.io/modules/u11d-com/medusajs/aws) as a reference.
+
+Example variables:
+
+```hcl
+backend_container_image = "your-medusa-backend-image:v1.20.11"
+project     = "medusa"
+environment = "prod"
+owner       = "my-team"
+```
+
+### Step 3: Initialize Terraform
+
+Run the following command to initialize Terraform and download the required providers:
+
+```shell
+terraform init
+```
+
+### Step 4: Plan the Infrastructure
+
+Preview the resources that Terraform will create:
+
+```shell
+terraform plan
+```
+
+### Step 5: Apply the Infrastructure
+
+Deploy the infrastructure by running:
+
+```shell
+terraform apply
+```
+
+Confirm the deployment by typing `yes` when prompted.
+
+### Step 6: Verify the Setup
+
+1. Check the Terraform outputs for the Medusa backend URL.
+2. Use the URL to verify that the backend is running.
+
+## Connecting Medusa to the Deployed Infrastructure
+
+Once the infrastructure is deployed, the Medusa backend URL will be available in the Terraform outputs. Use this URL to connect your Medusa application to the deployed backend.
+
+### Configuring Medusa with Provisioned AWS Services
+
+In addition to the backend, the Terraform module provisions key AWS resources, including an Amazon S3 bucket for object storage and an Amazon ElastiCache (Redis) instance. To integrate these services with your Medusa application:
+
+1. Configure S3 and Redis Plugins:
+
+- Set up the appropriate Medusa plugins for S3 and Redis to enable object storage and caching functionalities.
+  - [Redis Cache Module](https://docs.medusajs.com/v1/development/cache/modules/redis)
+  - [File Service - S3 plugin](https://docs.medusajs.com/v1/plugins/file-service/s3)
+- The same Redis instance can be utilized for multiple Medusa modules, including cache, event handling, file storage, and the workflow engine.
+
+2. Use Predefined Environment Variables:
+
+- The Terraform module automatically generates all necessary environment variables, following Medusa’s naming conventions.
+- These variables simplify the configuration process, ensuring seamless compatibility with Medusa.
+
+All internal connections between services are securely configured with proper access controls, reducing the need for additional security configurations.
+By following these steps, you’ll ensure your Medusa application is fully integrated with the cloud infrastructure, optimized for performance, scalability, and security.
+
```

Comment:
Add indentation to dash list items and remove the spacing between the numbered items and dashed items.